core/src/ast/types.rs

```diff
@@ -319,6 +319,16 @@ impl Mutability {
     }
 }
 
+/// For types like `Result`/`DiplomatResult`, `&[T]`/`DiplomatSlice<T>` which can be
+/// specified using (non-ffi-safe) Rust stdlib types, or FFI-safe `repr(C)` types from
+/// `diplomat_runtime`, this tracks which of the two were used.
+#[derive(Copy, Clone, PartialEq, Eq, Hash, Serialize, Deserialize, Debug)]
+#[allow(clippy::exhaustive_enums)] // This can only have two values
+pub enum StdlibOrDiplomat {
```

Comment:
below you have `get_diplomat_slice_type` and `get_rust_slice_type` methods, but here it's `StdLib` and `Diplomat`. align?